arch/arm64/src/common/arm64_gicv3.c

```diff
@@ -249,16 +249,6 @@ void arm64_gic_irq_enable(unsigned int intid)
   uint32_t mask = BIT(intid & (GIC_NUM_INTR_PER_REG - 1));
   uint32_t idx  = intid / GIC_NUM_INTR_PER_REG;
 
-  /* Affinity routing is enabled for Non-secure state (GICD_CTLR.ARE_NS
-   * is set to '1' when GIC distributor is initialized) ,so need to set
-   * SPI's affinity, now set it to be the PE on which it is enabled.
-   */
-
-  if (GIC_IS_SPI(intid))
-    {
-      arm64_gic_write_irouter((GET_MPIDR() & MPIDR_ID_MASK), intid);
```

Comment:
why remove the dynamic routing？

Reply:
Set SPI interrupt affinity routing default to CPU0 first, and keep them unchanged unless up_affinity_irq is called.

Reply:
We are using SPI to route to other CPUs, If you have special requirement please add a configuration option to disable routing to different CPUs， please set default value to n if smp case

Reply:
arm64_gic_irq_enable:The funtion should be only enable irq, 
set affinity should be call up_affinity_irq,

Reply:
So, this design will caused all kernel code must to call up_affinity_irq() before up_enable_irq()? Why is the behavior not which CPU is enabled and routing to which CPU to? Why don't you guys call up_affinity_irq(0) before up_enable_irq() to routing to CPU0?

Reply:
Hi @anchao,

This PR is to align with arm-v7a behavior and avoid changing affinity unexpectedly when enabling IRQ.
Driver may want to change the affinity by calling `up_affinity_irq`

https://github.com/apache/nuttx/blob/master/arch/arm/src/armv7-a/arm_gicv2.c#L208-L212


I'm unfamiliar with what kernel code requires, does arm-v7a have same problem?

Reply:
@XuNeo @lipengfei28 
https://github.com/apache/nuttx/pull/14172#discussion_r1797705346 

as I mentioned before that this feature could be add a new config, please review this PR https://github.com/apache/nuttx/pull/14235